lighthouse-core/gather/driver.js

```diff
@@ -696,15 +696,15 @@ class Driver {
   }
 
   setThrottling(flags, passConfig) {
-    const p = [];
-    if (passConfig.useThrottling) {
-      if (!flags.disableNetworkThrottling) p.push(emulation.enableNetworkThrottling(this));
-      if (!flags.disableCpuThrottling) p.push(emulation.enableCPUThrottling(this));
-    } else {
-      p.push(emulation.disableNetworkThrottling(this));
-      p.push(emulation.disableCPUThrottling(this));
-    }
-    return Promise.all(p);
+    const useCpuThrottling = passConfig.useThrottling && !flags.disableCpuThrottling;
+    const useNetworkThrottling = passConfig.useThrottling && !flags.disableNetworkThrottling;
+    const cpuThrottleFunc = `${useCpuThrottling ? 'enable' : 'disable'}CPUThrottling`;
+    const networkThrottleFunc = `${useNetworkThrottling ? 'enable' : 'disable'}NetworkThrottling`;
```

Comment:
string construction is pretty bad, but so was the original :P

Short of a refactor of `emulation.js`, what about
```js
setThrottlings(flags, passConfig) {
  const throttleCpu = passConfig.useThrottling && !flags.disableCpuThrottling;
  const cpuPromise = throttleCpu ? emulation.enableCPUThrottling(this) :
      emulation.disableCPUThrottling(this);

  const throttleNetwork = passConfig.useThrottling && !flags.disableNetworkThrottling;
  const networkPromise = throttleNetwork ? emulation.enableNetworkThrottling(this) :
      emulation.disableCPUThrottling(this);

  return Promise.all([cpuPromise, networkPromise]);
}
```